tsl/test/expected/dist_hypertable-12.out

```diff
@@ -2309,38 +2309,25 @@ UPDATE disttable_replicated SET device = 2 WHERE device = (SELECT device FROM de
-                       Sort Key: disttable_replicated_1.device
-                       ->  Sort (actual rows=0 loops=1)
-                             Output: disttable_replicated_1.device
-                             Sort Key: disttable_replicated_1.device
-                             Sort Method: quicksort 
-                             ->  Seq Scan on public.disttable_replicated disttable_replicated_1 (actual rows=0 loops=1)
-                                   Output: disttable_replicated_1.device
-                       ->  Foreign Scan on _timescaledb_internal._hyper_6_12_dist_chunk _hyper_6_12_dist_chunk_1 (actual rows=1 loops=1)
-                             Output: _hyper_6_12_dist_chunk_1.device
-                             Data node: data_node_1
-                             Remote SQL: SELECT device FROM _timescaledb_internal._hyper_6_12_dist_chunk ORDER BY device ASC NULLS LAST
-                       ->  Foreign Scan on _timescaledb_internal._hyper_6_13_dist_chunk _hyper_6_13_dist_chunk_1 (actual rows=1 loops=1)
-                             Output: _hyper_6_13_dist_chunk_1.device
-                             Data node: data_node_2
-                             Remote SQL: SELECT device FROM _timescaledb_internal._hyper_6_13_dist_chunk ORDER BY device ASC NULLS LAST
-                       ->  Foreign Scan on _timescaledb_internal._hyper_6_14_dist_chunk _hyper_6_14_dist_chunk_1 (actual rows=1 loops=1)
-                             Output: _hyper_6_14_dist_chunk_1.device
-                             Data node: data_node_3
-                             Remote SQL: SELECT device FROM _timescaledb_internal._hyper_6_14_dist_chunk ORDER BY device ASC NULLS LAST
-                       ->  Foreign Scan on _timescaledb_internal._hyper_6_15_dist_chunk _hyper_6_15_dist_chunk_1 (actual rows=1 loops=1)
-                             Output: _hyper_6_15_dist_chunk_1.device
-                             Data node: data_node_1
-                             Remote SQL: SELECT device FROM _timescaledb_internal._hyper_6_15_dist_chunk ORDER BY device ASC NULLS LAST
-                       ->  Foreign Scan on _timescaledb_internal._hyper_6_16_dist_chunk _hyper_6_16_dist_chunk_1 (actual rows=1 loops=1)
-                             Output: _hyper_6_16_dist_chunk_1.device
-                             Data node: data_node_2
-                             Remote SQL: SELECT device FROM _timescaledb_internal._hyper_6_16_dist_chunk ORDER BY device ASC NULLS LAST
-                       ->  Foreign Scan on _timescaledb_internal._hyper_6_17_dist_chunk _hyper_6_17_dist_chunk_1 (actual rows=1 loops=1)
-                             Output: _hyper_6_17_dist_chunk_1.device
-                             Data node: data_node_3
-                             Remote SQL: SELECT device FROM _timescaledb_internal._hyper_6_17_dist_chunk ORDER BY device ASC NULLS LAST
+                 ->  Custom Scan (AsyncAppend) (actual rows=1 loops=1)
```

Comment:
Could someone explain to me why queries involving insert/update are not optimized in distributed hypertables? It seems to me that this plan looks fine too. This is one regression I saw when rebasing on distributed hypertables branch.
@mkindahl @k-rus

Reply:
cc @erimatnor

Reply:
Yes, I think your explanation is correct. Then I will keep the test case like this.
Are you OK with the current version of the code that still does the marking during preprocessing for PG12 as well, or do you want it to be changed in this PR such that PG12 *only* does the marking of hypertables in the get_relation_hook? To change this, there'd need to be additional #ifdef's in preprocess_query that exclude PG12 specifically.

Reply:
For the sake of being cautious, I think we can refactor expansion handling for PG12 later if that is really what we'd like to do. Feel free to take a look at it though.

Reply:
Ok, will leave that for a future improvement then